src/components/ui/MonacoEditor.svelte

```diff
@@ -70,6 +71,9 @@
       value,
     };
     monaco = await import('monaco-editor');
+    monaco.languages.typescript.typescriptDefaults.setWorkerOptions({
+      customWorkerPath: '/customTS.worker.js',
```

Comment:
Instead of loading the worker in the `MonacoEditor` component, can we make the component more agnostic to the worker it's loading? For example, we want the [SequencingEditor](https://github.com/NASA-AMMOS/aerie-ui/blob/develop/src/components/sequencing/SequenceEditor.svelte) to load a custom worker to provide custom sequencing EDSL diagnostics. Can we do the custom worker configuration in that component instead of this component?

Reply:
I'm not sure we actually can move it. Looking at the implementation, it does seem that the worker setup is a singleton for each language. This would mean we need to inject this no matter where the worker is instantiated to ensure it gets applied on the first run and actually gets loaded. I plan to test this a bit more completely in a little, however I think we'll just need to make our custom worker not do anything by default and basically "register" a specific model (roughly an instance of the editor on the page) that trigger some override setup.